card/mod/core/set/all/update_read_rules.rb

```diff
@@ -6,25 +6,10 @@
   self.class.clear_rule_cache
 end
 
-def id_of_set_class set
-  set && (set_class = set.tag) && set_class.id
-end
-
 event :expire_related, :finalize do
-  subcards.keys.each do |key|
-    Card.cache.soft.delete key
-  end
-  expire # FIXME: where do we put this. Here it deletes @stage
-  reset_patterns
+  ActManager.expirees << key
+  # reset_patterns
   if is_structure?
-    structuree_names.each do |name|
-      Card.expire name
-    end
+    structuree_names.each { |name| ActManager.expirees << name }
   end
 end
-
-event :expire_related_names, before: :expire_related, changed: :name do
-  # FIXME: look for opportunities to avoid instantiating the following
-  descendants.each(&:expire)
-  name_referers.each(&:expire)
```

Comment:
Tests may prove me wrong, but I think this should be unnecessary now that descendants and name_referers are both handled in subcards (which will expire themselves).